consensus/src/test_utils/mock_state_computer.rs

```diff
@@ -108,24 +127,45 @@ impl StateComputer for MockStateComputer {
     }
 }
 
-pub struct EmptyStateComputer;
+pub struct EmptyStateComputer {
+    private_key: Ed25519PrivateKey,
+}
+
+impl EmptyStateComputer {
+    pub fn new() -> Self {
+        Self {
+            private_key: Ed25519PrivateKey::generate_for_testing(),
+        }
+    }
+}
 
 #[async_trait::async_trait]
 impl StateComputer for EmptyStateComputer {
     type Payload = TestPayload;
 
     fn compute(
         &self,
-        _block: &Block<Self::Payload>,
+        block: &Block<Self::Payload>,
         _parent_block_id: HashValue,
-    ) -> Result<StateComputeResult> {
-        Ok(StateComputeResult::new(
+    ) -> Result<StateComputeResultWithSignature> {
+        let result = StateComputeResult::new(
             *ACCUMULATOR_PLACEHOLDER_HASH,
             vec![],
             0,
+            vec![],
+            0,
             None,
             vec![],
             vec![],
+        );
+        let vote_proposal = VoteProposal::new(
+            result.extension_proof(),
+            block.clone(),
+            result.epoch_state().clone(),
+        );
+        Ok(StateComputeResultWithSignature::new(
+            result,
+            self.private_key.sign_message(&vote_proposal.hash()),
```

Comment:
if you made signatures optional, it would make some of these tests easier. could we add a commit to this pr or come up with a plan to get us there?

Reply:
I will use the optional field then this file should be intact.